### PR TITLE
IPv6 ingress node firewall intg/fixes and e2e support

### PR DESCRIPTION
**Pull request by msherif1234**
misc fixes to get ipv6 rules working
    - fix kernel patch to handle ipv6 packets using the right header
    - add ipv6 test case to e2e run
    - cleanup the protocol name for icmpv6
event logs from e2e run
```
2022/08/03 13:59:13     ruleId 10 action Drop len 118
2022/08/03 13:59:13     ipv6 src addr fc00:f853:ccd:e793::1 dst addr fc00:f853:ccd:e793::2
2022/08/03 13:59:13     icmpv6 type 128 code 0
2022/08/03 13:59:03     ruleId 10 action Drop len 98
2022/08/03 13:59:03     ipv4 src addr 172.18.0.1 dst addr 172.18.0.2
2022/08/03 13:59:03     icmpv4 type 8 code 0
```

### CODE REVIEW

**Comment by msherif1234** [comment]
https://github.com/kubernetes-sigs/kind/issues/2855

File: bpf/ingress_node_firewall_kernel.c [hunk 1]
```diff
@@ -51,14 +51,24 @@ struct {
...
+                  __u8 *icmpType, __u8 *icmpCode, __u8 is_v4) {
+
+    if (likely(is_v4)) {
+        struct iphdr *iph = dataStart;
+        dataStart += sizeof(struct iphdr);
+        if (unlikely(dataStart > dataEnd)) {
+            return -1;
+        }
+		*proto = iph->protocol;
+	} else {
```
Comment (martinkennelly) [inline bpf/ingress_node_firewall_kernel.c:63]: Pull this "else" back 4 spaces. Hard to read.

**Review by martinkennelly — COMMENTED** [review]
/lgtm
just fix the c code formatting

File: bpf/ingress_node_firewall_kernel.c (hunk 1 above)
Comment (martinkennelly) [reply bpf/ingress_node_firewall_kernel.c:63]: This may merge before you fix this because I put lgtm on it. You can follow up with a PR to format it.

Comment (msherif1234) [reply bpf/ingress_node_firewall_kernel.c:63]: oh sorry will fmt it Thanks for the review

**Pull request merged**